allennlp/tests/models/bidirectional_lm_test.py

```diff
@@ -25,42 +27,17 @@ def test_forward_pass_runs_correctly(self):
 
         # The model should preserve the BOS / EOS tokens.
         embeddings = result["lm_embeddings"]
-        assert tuple(embeddings.shape) == (2, 8, 14)
+        assert tuple(embeddings.shape) == self.expected_embedding_shape
 
         loss = result["loss"].item()
         forward_loss = result["forward_loss"].item()
         backward_loss = result["backward_loss"].item()
+        np.testing.assert_almost_equal(loss, (forward_loss + backward_loss) / 2,
+                                       decimal=3)
 
-        np.testing.assert_almost_equal(loss, (forward_loss + backward_loss) / 2, decimal=3)
-
-
-class TestBidirectionalLMUnsampled(TestBidirectionalLM):
+class TestBidirectionalLanguageModelUnsampled(TestBidirectionalLanguageModel):
     def setUp(self):
         super().setUp()
-        self.set_up_model(self.FIXTURES_ROOT / 'bidirectional_lm' / 'experiment_unsampled.jsonnet',
-                          self.FIXTURES_ROOT / 'bidirectional_lm' / 'sentences.txt')
-
-class TestBidirectionalLMTransformer(TestBidirectionalLM):
-    def setUp(self):
-        super().setUp()
-        self.set_up_model(self.FIXTURES_ROOT / 'bidirectional_lm' / 'experiment_transformer.jsonnet',
-                          self.FIXTURES_ROOT / 'bidirectional_lm' / 'sentences.txt')
-
-    def test_forward_pass_runs_correctly(self):
-        training_tensors = self.dataset.as_tensor_dict()
-        result = self.model(**training_tensors)
-
-        assert set(result) == {"loss", "forward_loss", "backward_loss",
-                               "lm_embeddings", "noncontextual_token_embeddings", "mask"}
-
-        # The model should preserve the BOS / EOS tokens.
-        embeddings = result["lm_embeddings"]
-        # The BidirectionalLanguageModelTransformer uses input size * 2 as the output size unlike
-        # a bidirectional LSTM, which uses hidden size * 2.
-        assert tuple(embeddings.shape) == (2, 8, 32)
```

Comment:
Something seems off here. Shouldn't the expected dimension be set to `(2, 8, 32)`? Sorry if I'm missing something.

Reply:
It's concerning that the tests didn't fail...

Reply:
So this experiment uses `experiment_bidirectional_unsampled.jsonnet`, which takes its config from `experiment_unsampled.jsonnet`. In that, we've got:

```
"contextualizer": {
        "type": "lstm",
        "bidirectional": true,
        "num_layers": 3,
        "input_size": 16,
        "hidden_size": 7,
    }
  },
```

So hidden size * 2 = 14?

Reply:
the `TestBidirectionalLMTransformer` was moved to a different class, which is why it isn't there.

Reply:
D'oh. Sorry! Read that diff all wrong.

Reply:
sorry it's so convoluted with stuff moving around everywhere, but also not completely disappearing...